stone/backend.py

```diff
@@ -40,28 +41,14 @@ def remove_aliases_from_api(api):
         # Remove nested aliases first. This way, when we replace an alias with
         # its source later on, it too is alias free.
         for alias in namespace.aliases:
-            data_type = alias
-            while True:
-                # For better or for worse, all non-user-defined types that
-                # reference other types do so with a 'data_type' attribute.
-                if hasattr(data_type, 'data_type'):
-                    if is_alias(data_type.data_type):
-                        # Skip the alias (this looks so terrible...)
-                        data_type.data_type = data_type.data_type.data_type
-                    data_type = data_type.data_type
-                else:
-                    break
-
+            # For better or for worse, all non-user-defined types that
+            # reference other types do so with a 'data_type' attribute.
+            if hasattr(alias, 'data_type'):
+                resolve_aliases(alias.data_type)
```

Comment:
It just needs to be `resolve_aliases(alias.data_type)`. Although looking at this again, I think it would be much better to pass in the parent `alias` obj like in your example (don't remember what my reasoning was for passing in `alias.data_type`).